File: book/python_algorithm_interview/Q/10.ts
function arraypartition(nums: number[]): number {
  nums.sort();
  let result = 0;

  for (let i = 0; i < nums.length; i++) {
    if (i % 2 === 0) {
      result += nums[i];
    }
  }

  return result;
}

console.log(arraypartition([1, 4, 3, 2]));

File: book/python_algorithm_interview/Q/10.py
from typing import List


def list_partition(nums: List[int]) -> int:
    nums.sort()
    result = 0
    for i, v in enumerate(nums):
        if i % 2 == 0:
            result += v
    return result


def arrayPairSumPythonic(nums: List[int]) -> int:
    return sum(sorted(nums)[::2])


print(arrayPairSumPythonic([1, 4, 3, 2]))
